frontend/src/pages/GraphicalSelector/GraphicalSelector.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect, useRef } from "react";
+import { useSelector } from "react-redux";
+import axios from "axios";
+import G6 from "@antv/g6";
+import { findDOMNode } from "react-dom";
+
+const GraphicalSelector = () => {
+  // eslint-disable-next-line
+  const { programCode, majors, minor } = useSelector((state) => state.degree);
+
+  const [courses, setCourses] = useState([]);
+  const [courseEdges, setCourseEdges] = useState([]);
+  const ref = useRef(null);
+  let graph = null;
+  const loadGraph = () => {
+    if (!graph) {
+      graph = new G6.Graph({
+        // eslint-disable-next-line react/no-find-dom-node
+        container: findDOMNode(ref.current),
+        width: 1920,
+        height: 870,
+        modes: {
+          default: ["drag-canvas", "drag-node"],
```

Comment:
In terms of cleaning up distribution of nodes, does the `collapse-expand` conflict with the draggability?
If not, allowing the user to collapse nodes dependent on a current node could be pretty useful for decluttering.

Reply:
yea - im not so sure if it works with this preset. putting it there didnt really do much.